gson/src/main/java/com/google/gson/Gson.java

```diff
@@ -932,98 +954,165 @@ public void toJson(JsonElement jsonElement, JsonWriter writer) throws JsonIOExce
    * or if {@code json} is empty.
    * @throws JsonSyntaxException if json is not a valid representation for an object of type
    * classOfT
+   *
+   * @see #fromJson(Reader, Class)
+   * @see #fromJson(String, TypeToken)
    */
   public <T> T fromJson(String json, Class<T> classOfT) throws JsonSyntaxException {
-    Object object = fromJson(json, (Type) classOfT);
+    T object = fromJson(json, TypeToken.get(classOfT));
```

Comment:
I was thinking this was introducing an object allocation where there wasn't one before, but actually the code already does `TypeToken.get(...)` later on.